src/app/editor/contextualListener.js

```diff
@@ -94,6 +91,43 @@ class ContextualListener {
     }
   }
 
+  _getGasEstimation (results) {
+    this.contractName = Object.keys(results.data.contracts[results.source.target])[0]
+    this.target = results.data.contracts[results.source.target]
+    this.contract = this.target[this.contractName]
+    this.estimationObj = this.contract.evm.gasEstimates
+    if (this.estimationObj.external) this.externalFunctions = Object.keys(this.estimationObj.external)
+    if (this.estimationObj.internal) this.internalFunctions = Object.keys(this.estimationObj.internal)
+    this.creationCost = this.estimationObj.creation.totalCost
+    this.codeDepositCost = this.estimationObj.creation.codeDepositCost
+  }
+
+  gasEstimation (node) {
+    if (node.name === 'FunctionDefinition') {
+      if (!node.attributes.isConstructor) {
+        var functionName = node.attributes.name
+        if (this.externalFunctions) {
+          return {executionCost: this.estimationObj.external[this._getFn(this.externalFunctions, functionName)]}
```

Comment:
Instead of looking up the function in both external and internal, it should make use of the node attributes to decide which one it is.

Also the external lookup will not work because that has function signature, function name + arguments.

Reply:
Commited some changes, pls check if that's what you had in mind @axic

Reply:
we still need to better search for the function signature, cause `functionName` does not contain it.

Reply:
@yann300 What do you mean by this exactly?

Reply:
that `functionName` would be `get` for instance but in `this.externalFunctions` you will have `get(uint _v)`

Reply:
I see. It works now, right? It's just not very clean and elegant...

Reply:
Any suggestions?

Reply:
We could maybe rather use split? Let me try